elasticsearch/index.d.ts

```diff
@@ -208,242 +369,1204 @@ declare module Elasticsearch {
+        type?: NameList;
+    }
+
+    export interface MSearchResponse<T> {
+        responses?: SearchResponse<T>[];
+    }
+
+    export interface MSearchTemplateParams extends GenericParams {
+        search_type?: "query_then_fetch" | "query_and_fetch" | "dfs_query_then_fetch" | "dfs_query_and_fetch";
+        index?: NameList;
+        type?: NameList;
+    }
+
+    export interface MTermVectorsParams extends GenericParams {
+        ids?: NameList;
+        termStatistics?: boolean;
+        fieldStatistics?: boolean;
+        fields?: NameList;
+        offsets?: boolean;
+        positions?: boolean;
+        payloads?: boolean;
+        preference?: string;
         routing?: string;
-        timeout?: Date | number;
-        timestamp?: Date | number;
+        parent?: string;
+        realtime?: boolean;
         version?: number;
-        versionType?: string;
+        versionType?: VersionType;
+        index: string;
+        type: string;
+    }
+
+    export interface PingParams extends GenericParams {
+    }
+
+    export interface PutScriptParams extends GenericParams {
+        id: string;
+        lang: string;
+        body: any;
+    }
+
+    export interface PutTemplateParams extends GenericParams {
+        id: string;
+        body: any;
+    }
+
+    export interface ReindexParams extends GenericParams {
+        refresh?: boolean;
+        timeout?: DateLike;
+        waitForActiveShards?: string;
+        waitForCompletion?: boolean;
+        requestsPerSecond?: number;
+        body: {
+            conflicts?: string;
+            source: {
+                index: string | string[];
+                type?: string | string[];
+                query?: any;
+                sort?: any;
+                size?: number;
+                remote?: {
+                    host: string;
+                    username?: string;
+                    password?: string;
+                }
+            };
+            dest: {
+                index: string;
+                version_type?: string;
+                op_type?: string;
+                routing?: string;
+                pipeline?: string;
+            };
+            script?: {
+                inline: string;
+                lang: string;
+            }
+        }
+    }
+
+    export interface ReindexResponse {
+        took: number;
+        updated: number;
+        created: number;
+        batches: number;
+        version_conflicts: number;
+        retries: {
+            bulk: number;
+            search: number;
+        }
+        throttled_millis: number;
+        failures: any[];
+    }
+
+    export interface ReindexRethrottleParams extends GenericParams {
+        requestsPerSecond: number;
+        taskId: string;
+    }
+
+    export interface RenderSearchTemplateParams extends GenericParams {
+        id: string;
     }
 
     export interface ScrollParams extends GenericParams {
-        scroll: string;
+        scroll: DateLike;
```

Comment:
The elasticsearch 5 docs shows a snippet for the scroll parameter, where it is clearly a string  of duration
```
var allTitles = [];

// first we do a search, and specify a scroll timeout
client.search({
  index: 'myindex',
  // Set to 30 seconds because we are calling right back
  scroll: '30s',
  search_type: 'scan',
  fields: ['title'],
  q: 'title:test'
```